### PR TITLE
Fix matrix-free deprecation warnings

### PR DESCRIPTION
**Pull request by zjiaqi2018**
Fix matrix-free deprecation warnings, and it passed the test nsinker_gmg_idr. @tjhei

### CODE REVIEW

File: source/simulator/stokes_matrix_free.cc [hunk 1]
```diff
@@ -721,10 +721,18 @@ namespace aspect
 
         velocity.reinit (cell);
         velocity.read_dof_values (src.block(0));
+#if DEAL_II_VERSION_GTE(10,0,0)
+        velocity.evaluate (EvaluationFlags::gradients);
```
Comment (tjhei) [inline source/simulator/stokes_matrix_free.cc:725]: Is this new in 10 or already in 9.3?

**Comment by tjhei** [comment]
Fixes #4068

File: source/simulator/stokes_matrix_free.cc (hunk 1 above)
Comment (zjiaqi2018) [reply source/simulator/stokes_matrix_free.cc:725]: The EvaluationFlags is in 9.3.
Change to DEAL_II_VERSION_GTE(9.3.0)?

Comment (tjhei) [reply source/simulator/stokes_matrix_free.cc:725]: Yes, exactly. Because starting with 9.3 we can and should use the new method.

Comment (zjiaqi2018) [reply source/simulator/stokes_matrix_free.cc:725]: Done!

**Review by gassmoeller — APPROVED** [review]
Looks ready to go. Thanks @zjiaqi2018!

**Comment by gassmoeller** [comment]
Just waiting for the tester (github actions required approval, because it considers you a first time contributor).

**Pull request merged**